book/sending_emails.rst

```diff
@@ -0,0 +1,226 @@
+            $email->addAttachment($attachment);
+
+            // ...
+        }
+    }
+
+.. _book-sending-emails-builder:
+
+
+The ``EmailModelBuilder`` Class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+An alternative approach to manually creating ``Email`` models is to use the
+:class:`Oro\\Bundle\\EmailBundle\\Builder\\EmailModelBuilder` helper class which offers several
+methods to create new e-mails based on existing data:
+
+:method:`Oro\\Bundle\\EmailBundle\\Builder\\EmailModelBuilder::createEmailModel`
+    Create an e-mail as a copy of an existing e-mail.
+
+:method:`Oro\\Bundle\\EmailBundle\\Builder\\EmailModelBuilder::createReplyEmailModel`
+    Create an e-mail that is a response to an existing e-mail.
+
+:method:`Oro\\Bundle\\EmailBundle\\Builder\\EmailModelBuilder::createReplyAllEmailModel`
+    Create an e-mail that is a response to all recipients and the sender of an existing e-mail.
+
+:method:`Oro\\Bundle\\EmailBundle\\Builder\\EmailModelBuilder::createForwardEmailModel`
+    Create a new e-mail that forwards an existing e-mail.
+
+After e-mails have been processed (see below), they will be persisted to the database. You can
+create an e-mail model based on such a persisted entity, by using the useful
+:class:`Oro\\Bundle\\EmailBundle\\Builder\\EmailModelBuilder` helper class:
+
+.. code-block:: php
+    :linenos:
+
+    // src/Acme/DemoBundle/Controller/EmailController.php
+    namespace Acme\DemoBundle\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+    class EmailController extends Controller
+    {
+        public function sendMailAction()
+        {
+            // ...
+        }
+    }
+
+Sending the Mail
+----------------
+
+When you created your e-mail model, you can use the integrated mailer processor which is
+responsible for sending the e-mail and persisting it to the database (which also creates the needed
+contexts to customers, users, and so on):
+
+.. code-block:: php
+    :linenos:
+
+    // src/Acme/DemoBundle/Controller/EmailController.php
+    namespace Acme\DemoBundle\Controller;
+
+    // ...
+
+    class EmailController extends Controller
+    {
+        public function sendMailAction()
+        {
+            // ...
+            $email = ...;
+
+            $processor = $this->get('oro_email.mailer.processor');
+            $processor->process($email);
+        }
+    }
+
+When calling the :method:`Oro\\Bundle\\EmailBundle\\Mailer\\Processor::process` the mailer
+processor performs the following steps:
+
+#. It creates a new ``\Swift_Message`` instance and populate it with the data from your ``Email``
+   object.
+
+#. If you did not pass an ``EmailOrigin`` object which should be used to associate the mail in the
+   user interface with, the processor will create on based on the sender address and the selected
+   organization.
+
+#. The e-mail will be sent based on your application's `SwiftMailer configuration`_.
```

Comment:
You can additionally describe that email can be send by the user SMTP configuration (user profile) or system/organization SMTP configuration if they configured.